### ISSUE TITLE
High memory consumption for Pointcloud class

### ISSUE THREAD

**Issue by erezposner** [gradslam/gradslam#28]
Hi, running the example provided here:
https://gradslam.readthedocs.io/en/latest/tutorials/pointfusion_tutorial.html

I've noticed high memory consumption using Pointcloud class. Just an instance creation using this line
```
slam = PointFusion(device=device)
```
consumes 900MB of GPU  memory.

Also, during each SLAM step, approx size of point clouds (consists RGB, xyz and normals) of 3MB consumes approx 300MB of gpu memory.

I've tried using the inplace flag but this is an issue.
Can you please advise?
Thank you

**Comment by saryazdi** [comment]
Hi @erezposner,

The 900 MB of GPU memory consumption that you mentioned is due to the CUDA context that will be created on the device when the first tensor is created. This is related to pytorch, not gradslam. E.g. running the following code will also allocate 900 MB of GPU memory:
```
import torch
torch.tensor([0], device="cuda")
```

In regards to the memory consumption of SLAM, that has to do with the fact that the pointcloud gets larger with every iteration, and it also could be due to the gradients in the graph. So perhaps turning gradients off for any variables where you won't need them would help. Also running SLAM for shorter sequences would help.

**Comment by erezposner** [comment]
Well, This is not accurate.

It seems that there is a lot of unused information that is being stored in the cache and clearing it alleviate this issue without harming training.


Simply add `torch.cuda.empty_cache()` in icpslam.py.

```        if self.odom in ["icp", "gradicp"]:
            live_frame.poses = prev_frame.poses
            frames_pc = downsample_rgbdimages(live_frame, self.dsratio)
            pc2im_bnhw = find_active_map_points(pointclouds, prev_frame)
            maps_pc = downsample_pointclouds(pointclouds, pc2im_bnhw, self.dsratio)
            pc2im_bnhw=[]
            transform = self.odomprov.provide(maps_pc, frames_pc)
            torch.cuda.empty_cache()
```
